diff --git a/node_modules/react-native-android-widget/android/src/main/java/com/reactnativeandroidwidget/RNWidgetImageProvider.java b/node_modules/react-native-android-widget/android/src/main/java/com/reactnativeandroidwidget/RNWidgetImageProvider.java
index 1234567..abcdefg 100644
--- a/node_modules/react-native-android-widget/android/src/main/java/com/reactnativeandroidwidget/RNWidgetImageProvider.java
+++ b/node_modules/react-native-android-widget/android/src/main/java/com/reactnativeandroidwidget/RNWidgetImageProvider.java
@@ -76,8 +76,9 @@ public class RNWidgetImageProvider extends ContentProvider {
 
     private static void deleteImages(Context context, String prefix) {
         File folder = getFolderWithImages(context);
-        File[] files = folder.listFiles(pathname -> pathname.getName().startsWith(prefix));
+        if (!folder.exists()) return;
 
+        File[] files = folder.listFiles(pathname -> pathname.getName().startsWith(prefix));
         for (File f : Objects.requireNonNull(files)) {
             f.delete();
         }
